archive/extract.go

```diff
@@ -25,6 +25,12 @@ type PathMode struct {
 
 // Extract reads all entries from TarReader and extracts them to the filesystem
 func Extract(tr TarReader) error {
+	currentUmask := SetUmask(0)
```

Comment:
This affects reentrancy of `Extract`. Consumers may expect to call this method in parallel with other invocations.

Reply:
@sclevine updated in 1a3669d - what do you think?

Reply:
Sorry, it's really cf27827 through 1a3669d...